indexer-app/src/cmr/indexer/data/index_set_generics.clj

```diff
@@ -78,6 +92,41 @@
         (assoc (keyword index-name) converted-mapping)
         (assoc (keyword index-name-lower) converted-mapping))))
 
+ (defn get-settings
+   "Get the elastic settings from the configuration files. If the default number of shards has
+    changed, then use that instead of what was configured."
+   [index-definition]
+   (if-let [settings (:IndexSetup index-definition)]
+     (let [config-shards (get-in settings [:index :number_of_shards])
+           ;; A changed environment variable takes precidence, then the config file, then the default.
+           ;; If the the environment variable = default value then the environment variable is not set.
+           ;; If the enviornment variable is set, then use it.
+           num-shards (if (= (elastic-generic-index-num-shards) default-generic-index-num-shards)
+                        (if config-shards 
+                          (get-in settings [:index :number_of_shards])
+                          default-generic-index-num-shards)
+                        (elastic-generic-index-num-shards))]
```

Comment:
I think you may be able to use a cond here instead of the nested if. Something like:

(cond
    (= (elastic-generic-index-num-shards) default-generic-index-num-shards) (elastic-generic-index-num-shards)
    (config-shards) (get-in settings [:index :number_of_shards])
    :else default-generic-index-num-shards))

...but i'm sort of split on whether that is actually more readable in this case.

Reply:
I have updated the code to use cond.